components/announcement_bar/default_announcement_bar/announcement_bar.tsx

```diff
@@ -17,39 +16,37 @@ import ToggleModalButtonRedux from 'components/toggle_modal_button_redux';
 
 import {trackEvent} from 'actions/telemetry_actions.jsx';
 
-export default class AnnouncementBar extends React.PureComponent {
-    static propTypes = {
-        showCloseButton: PropTypes.bool,
-        color: PropTypes.string,
-        textColor: PropTypes.string,
-        type: PropTypes.string,
-        message: PropTypes.node.isRequired,
-        handleClose: PropTypes.func,
-        announcementBarCount: PropTypes.number.isRequired,
-        showModal: PropTypes.bool,
-        modalButtonText: PropTypes.string,
-        modalButtonDefaultText: PropTypes.string,
-        showLinkAsButton: PropTypes.bool,
-        warnMetricStatus: PropTypes.object,
-        isTallBanner: PropTypes.bool,
-        actions: PropTypes.shape({
-            incrementAnnouncementBarCount: PropTypes.func.isRequired,
-            decrementAnnouncementBarCount: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    showCloseButton: boolean;
+    color: string;
+    textColor: string;
+    type: string;
+    message: React.ReactNode;
+    handleClose?: (e?:any)=>void;
+    announcementBarCount?: number;
+    showModal?: ()=> void;
+    modalButtonText?: string;
+    modalButtonDefaultText?: string;
+    showLinkAsButton: boolean;
+    warnMetricStatus?: Record<string, unknown>;
+    isTallBanner: boolean;
+    actions: {
+        incrementAnnouncementBarCount: ()=>void;
+        decrementAnnouncementBarCount: ()=>void;
+    };
+}
 
+export default class AnnouncementBar extends React.PureComponent<Props> {
     static defaultProps = {
         showCloseButton: false,
         color: '',
         textColor: '',
         type: AnnouncementBarTypes.CRITICAL,
-        handleClose: null,
-        onButtonClick: null,
         showLinkAsButton: false,
         isTallBanner: false,
     }
 
-    componentDidMount() {
+    public componentDidMount() {// public for testing
```

Comment:
```suggestion
    // "public" due to this function being used in the test file
    public componentDidMount() {
```
I'd make the comment above the function and make it more clear

Reply:
These methods actually did not need to be public.